session_ii/javascript-fundamentals/session2/homework/mpricu/function.js

```diff
@@ -0,0 +1,79 @@
+
+const pro1 = new Promise((resolve, reject) => {
+    // setTimeout(() => resolve(1), 1000);
+    reject(new Error("error1"));
+});
+
+const pro2 = new Promise((resolve, reject) => {
+    // setTimeout(() => resolve(2), 400);
+    reject(new Error("error2"));
+});
+
+const pro3 = new Promise((resolve, reject) => {
+    // setTimeout(() => resolve(3), 300);
+    reject(new Error("error3"));
+});
+
+const pro4 = new Promise((resolve, reject) => {
+    // setTimeout(() => resolve(4), 100);
+    reject(new Error("error4"));
+});
+
+const pro5 = new Promise((resolve, reject) => {
+    // setTimeout(() => resolve(5), 200);
+    reject(new Error("error5"));
+});
+
+promises = [pro1, pro2, pro3, pro4, pro5];
+
+const erra = [];
+
+// function to complete Promise.race in case all the promises rejected
+function allReject(timeout) {
```

Comment:
not sure I understand the purpose of this function (beside wanting to show that you love promises of course ❤️ ) :-)

Reply:
If all of the child promises reject you will wait till the end of the world for the Promise.race call to complete. So you have to have another promise which will be resolved after a given amount of time.
The rejection value  of that promise was in fact longer (too long!): "ihatepromisesbecauseilovefacts"....nothing to do with the JavaScript promises....:))